R/special_orthogonal_group.R

```diff
@@ -74,6 +74,55 @@ SpecialOrthogonalGroup <- setRefClass("SpecialOrthogonalGroup",
                                           }
                                           stopifnot(length(dim(regularized.point)) == 2)
                                           return(regularized.point)
+                                        },
+
+                                        SkewMatrixFromVector = function(vec){
+                                          "In 3D, compute the skew-symmetric matrix,
+                                          known as the cross-product of a vector,
+                                          associated to the vector vec."
+                                          vec <- ToNdarray(vec, to.ndim = 2)
+                                          n.vecs <- dim(vec)[1]
+                                          vec.dim <- dim(vec)[2]
+                                          if (.self$n == 3) {
+                                            levi.civita.symbol <- array(data = c(
+                                              c(0, 0, 0),
+                                              c(0, 0, 1),
+                                              c(0, -1, 0),
+                                              c(0, 0, -1),
+                                              c(0, 0, 0),
+                                              c(1, 0, 0),
+                                              c(0, 1, 0),
+                                              c(-1, 0, 0),
+                                              c(0, 0, 0)
+                                            ), dim = c(3, 3, 3)
+                                            )
+
+                                            basis.vec.1 <- array(c(1, 0, 0) * n.vecs)
+                                            basis.vec.2 <- array(c(0, 1, 0) * n.vecs)
+                                            basis.vec.3 <- array(c(0, 0, 1) * n.vecs)
+
+                                            basis.vec.1.array <- ToNdarray(basis.vec.1, to.ndim = 2)
+                                            intermediate <- do.call(cbind,lapply(seq_len(3),function(i) basis.vec.1.array %*% levi.civita.symbol[i,,]))
+                                            intermediate <- array(intermediate, dim = c(3,3))
+                                            cross.prod.1 <- vec %*% intermediate
+
+                                            basis.vec.2.array <- ToNdarray(basis.vec.2, to.ndim = 2)
+                                            intermediate <- do.call(cbind,lapply(seq_len(3),function(i) basis.vec.2.array %*% levi.civita.symbol[i,,]))
+                                            intermediate <- array(intermediate, dim = c(3,3))
```

Comment:
intermediate is not of shape (3, 3) already? Could we make it directly (3, 3), instead of "correcting" it here?

Using `lbind` maybe?